src/Build/Logging/FileLogger.cs

```diff
@@ -86,6 +86,8 @@ private void InitializeFileLogger(IEventSource eventSource, int nodeCount)
 
             try
             {
+                Directory.CreateDirectory(Path.GetDirectoryName(Path.GetFullPath(_logFileName)));
```

Comment:
I can't think of a case when `GetFullPath()` would return something bad or if `GetDirectoryName()` would return null but I think I'd prefer to see some more guarding around these calls.  Either check some of the return values or put this line in its own `try...catch` which ignores the exception and lets the exception come from `FileUtilities.OpenWrite()` instead.